lambdaclass/game_backend
# [GH-1215] change integers to floats for position representations

closes [#1215](https://github.com/lambdaclass/curse_of_myrra/issues/1215)

# Motivation

We want to use floats to represent positions in both the frontend and the backend .

# Changes Made

Changed to float all fields related to:

- size (this includes the size of the characters, loots, projectiles, map size, map modifications values, etc.)
- position
- speed (both character and projectile speed)
- attack ranges

Also changed the config to divide by 100 all values related to distances. That is speeds, ranges and zone updates. This change is also replicated in the client. Before we had to do this because the game's board was 100x100 units in the client but 10000x10000 units in the backend. We couldn't use 100x100 in the backend with integers because the tiles would be too big and the movement would end up being too choppy. Now that we have floats we can just use 100x100 map size and have movement speeds with decimal amounts.

There's a small warning with this as using floats might incurr in precision errors or small desyncs between client and server ([not all computers represent and handle floats in a consistent way, and it might also vary based on the compiler used](https://grouper.ieee.org/groups/msc/ANSI_IEEE-Std-754-2019/background/addendum.html)), but for now unless we see playability issues we'll deem it non-relevant

# Steps to test this

You can use the branch `use-floats-for-backend-client-comm` from the [curse_of_myrra repo](https://github.com/lambdaclass/curse_of_myrra)

Check that the game still feels ok, particularly projectiles, hitboxes, movement and the zone
